### DISCUSSION TITLE
Does OpenThread support dual interfaces, Thread and Ethernet?

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#10908] [Q&A]
We want to develop a product based on CoAP. Its CoAP resources support access from both Ethernet and Thread networks. Does the OpenThread stack support dual interfaces? Which chip can implement this solution?

**Comment by jinpeng1989** [comment]
Both single-chip and multi-chip solutions are available. 
![image](https://github.com/user-attachments/assets/ee6d10a1-6bc6-4d5c-920d-f2b39371aeac)

**Comment by jwhui** [comment]
OpenThread currently does not support multiple interfaces on its own. Existing implementations typically involve integrating OpenThread as a network interface into another network stack (e.g. Linux network stack, Zephyr network stack, LwIP, etc.).

**Comment by jinpeng1989** [comment]
In the NCP solution, the Host Processor and 802.15.4 SoC communicate over the Spinel protocol. If the operating system of the Host processor is free-RTOS, and the Third-Party IPv6 stack is lwip, what components should be used to implement Spinel of Host processor?
![image](https://github.com/user-attachments/assets/ef0569f2-4fe3-4c77-9d31-a0f309589a7d)

**Reply by jwhui** [reply]
Currently, the OpenThread project itself does not include any integration with FreeRTOS + LwIP host. That said, I know that implementations exist, such as with Espressif.

**Comment by jinpeng1989** [comment]
Thank you.
What Spinel commands do Host Processor and NCP use to exchange IP packets? For example, sniffing raw packets are passed by NCP to Host via **CMD_VALUE_IS: PROP_STREAM_RAW: ...**. Can you point it out for me?

**Reply by abtink** [reply]
Check these two: 
- [`SPINEL_PROP_STREAM_NET`](https://github.com/openthread/openthread/blob/3702c88a76486fa04448e578158c25390a9c6599/src/lib/spinel/spinel.h#L3476-L3502) and  
- [`SPINEL_PROP_STREAM_NET_INSECURE`](https://github.com/openthread/openthread/blob/3702c88a76486fa04448e578158c25390a9c6599/src/lib/spinel/spinel.h#L3503-L3529)

**Comment by jinpeng1989** [comment]
Thank you.
